src/components/Button/Button.tsx

```diff
@@ -0,0 +1,53 @@
+import { ButtonHTMLAttributes } from "react";
+import { buttonLayout } from "./Button.style";
+
+type ButtonStyleProps = "primary" | "secondary" | "ghost";
+
+interface ButtonProps
+  extends Omit<ButtonHTMLAttributes<HTMLButtonElement>, "style"> {
+  onClick: () => void;
+  children: React.ReactNode;
+  style?: ButtonStyleProps;
+  size?: "sm" | "full";
+  dataTestid?: string;
+}
```

Comment:
componentProps를 활용해도 괜찮을거 같아!

```suggestion
interface ButtonProps extends ComponentProps<'button'> {
  children: React.ReactNode;
  style?: ButtonStyleProps;
  size?: "sm" | "full";
  dataTestid?: string;
}
```

Reply:
그리고 이 컴포넌트는 공용 컴포넌트인가??